applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -306,7 +305,8 @@ class FEASTEigensystemSolver
             "Manually defined e_mid_re, e_mid_im, e_r are not used for real symmetric matrices" << std::endl;
     }
 
-    void CheckParameters(std::complex<double> T)
+    template<typename TScalar, typename std::enable_if<std::is_same<std::complex<double>, TScalar>::value, int>::type = 0>
```

Comment:
I think a better solution to `enable_if` is [explicit (full) template specialization](https://en.cppreference.com/w/cpp/language/template_specialization)

Reply:
to my understanding, I must then hardcode the ublas spaces in the specializations. Is this not contrary to the idea of templating the space? Please correct me, if I'm wrong